app/forms/openid_connect_logout_form.rb

```diff
@@ -13,24 +14,32 @@ class OpenidConnectLogoutForm
 
   RANDOM_VALUE_MINIMUM_LENGTH = OpenidConnectAuthorizeForm::RANDOM_VALUE_MINIMUM_LENGTH
 
-  validates :id_token_hint, presence: true
+  validates :client_id,
+            presence: { message: 'client_id is missing' },
+            if: -> { reject_id_token_hint? }
+  validates :id_token_hint, presence: true, unless: -> { reject_id_token_hint? }
+  validates :id_token_hint,
+            absence: {
+              message: ->(_, _) { I18n.t('openid_connect.logout.errors.id_token_hint_present') },
+            },
+            if: -> { reject_id_token_hint? }
```

Comment:
we can pass symbols to `:if` , might clear up some line noise

```suggestion
            if: :reject_id_token_hint?
  validates :id_token_hint, presence: true, unless: :reject_id_token_hint?
  validates :id_token_hint,
            absence: {
              message: ->(_, _) { I18n.t('openid_connect.logout.errors.id_token_hint_present') },
            },
            if: :reject_id_token_hint?
```

Reply:
thanks good catch!